### PR TITLE
Lps 72895

### PR DESCRIPTION
**Pull request by shuyangzhou**

### CODE REVIEW

File: modules/apps/static/portal-configuration/portal-configuration-persistence-test/src/testIntegration/java/com/liferay/portal/configuration/persistence/listener/test/ConfigurationModelListenerTest.java [hunk 1]
```diff
@@ -222,13 +222,13 @@ public void onBeforeSave(
...
 
-	private static Configuration _getConfiguration(String pid, String location)
+	private static Configuration _getConfiguration(String pid)
 		throws IOException {
 
 		return OSGiServiceUtil.callService(
 			_bundleContext, ConfigurationAdmin.class,
 			configurationAdmin -> configurationAdmin.getConfiguration(
-				pid, location));
+				pid, StringPool.QUESTION));
```
Comment (drewbrokke) [inline modules/apps/static/portal-configuration/portal-configuration-persistence-test/src/testIntegration/java/com/liferay/portal/configuration/persistence/listener/test/ConfigurationModelListenerTest.java]: @shuyangzhou for my education, what does this change accomplish?

File: modules/apps/static/portal-configuration/portal-configuration-persistence-test/src/testIntegration/java/com/liferay/portal/configuration/persistence/listener/test/ConfigurationModelListenerTest.java [hunk 2]
```diff
@@ -51,21 +52,10 @@
...
-			// persistence delete method throws, the configuration will still
-			// contain an internal field called 'deleted' that is set to true.
-			// This causes the configuration to throw an IllegalStateException
-			// if any other persistence methods are called on it afterwards.
-			// If the 'onBeforeDelete' method throws, this state occurs. To
-			// work around it for this test, we are catching the
-			// IllegalStateException during clean up.
-
-		}
+		ReflectionTestUtil.invoke(delegatee, "delete", new Class<?>[0]);
```
Comment (drewbrokke) [inline modules/apps/static/portal-configuration/portal-configuration-persistence-test/src/testIntegration/java/com/liferay/portal/configuration/persistence/listener/test/ConfigurationModelListenerTest.java]: This leads to the question of how do we truly deal with this in an actual use-case and not just in a test?  I'm not sure of the answer.  I sent a [pull request](https://github.com/apache/felix/pull/113) to Apache Felix, but even if they merge it I'm not sure what that process looks like for us to use that.

Is there any way to work around this issue?  Or is it just a bug in waiting?

File: modules/apps/static/portal-configuration/portal-configuration-persistence-test/src/testIntegration/java/com/liferay/portal/configuration/persistence/listener/test/ConfigurationModelListenerTest.java (hunk 1 above)
Comment (shuyangzhou) [inline modules/apps/static/portal-configuration/portal-configuration-persistence-test/src/testIntegration/java/com/liferay/portal/configuration/persistence/listener/test/ConfigurationModelListenerTest.java]: When we create a method, no matter it is public or not, we want to make sure the parameters are really used for something. In this case, the "location" is not really used, as we always pass in "?". Then we should not have this parameter, just use "?" directly.

File: modules/apps/static/portal-configuration/portal-configuration-persistence-test/src/testIntegration/java/com/liferay/portal/configuration/persistence/listener/test/ConfigurationModelListenerTest.java (hunk 2 above)
Comment (shuyangzhou) [inline modules/apps/static/portal-configuration/portal-configuration-persistence-test/src/testIntegration/java/com/liferay/portal/configuration/persistence/listener/test/ConfigurationModelListenerTest.java]: There is a real solution for this. It is apparently a design issue inside ConfigAdmin implementation. The memory state transition must be synchronized with DB access in a transactional way.

File: modules/apps/static/portal-configuration/portal-configuration-persistence-test/src/testIntegration/java/com/liferay/portal/configuration/persistence/listener/test/ConfigurationModelListenerTest.java (hunk 1 above)
Comment (drewbrokke) [inline modules/apps/static/portal-configuration/portal-configuration-persistence-test/src/testIntegration/java/com/liferay/portal/configuration/persistence/listener/test/ConfigurationModelListenerTest.java]: Oh, haha, I was reading the change as the opposite at the time.  It was a long Friday.

**Pull request closed without merging**